ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/InitializeMigrationCommand.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.tools.migrations.commands;
+
+import com.github.rvesse.airline.annotations.Command;
+import io.confluent.ksql.properties.PropertiesUtil;
+import io.confluent.ksql.rest.client.KsqlRestClient;
+import io.confluent.ksql.rest.client.RestResponse;
+import io.confluent.ksql.rest.entity.KsqlEntityList;
+import io.confluent.ksql.tools.migrations.MigrationConfig;
+import java.io.File;
+import java.util.Collections;
+import java.util.Optional;
+
+@Command(
+    name = "initialize",
+    description = "Initializes the schema metadata (stream and table)."
+)
+public class InitializeMigrationCommand extends BaseCommand {
+
+  private String createEventStream(final String name, final String topic, final String replicas) {
+    return "CREATE STREAM " + name + " (\n"
+        + "  version_key  STRING KEY,\n"
+        + "  version      STRING,\n"
+        + "  name         STRING,\n"
+        + "  state        STRING,  \n"
+        + "  checksum     STRING,\n"
+        + "  started_on   STRING,\n"
```

Comment:
Thanks for bringing this up. I think it's safe to assume customers running older Confluent Platform versions be interested in using this tool. For example, I think it's safe to assume someone upgrading to Confluent Platform 6.1.0 this month will try to use the migrations tool when it is released from the standalone distribution. We can and should be more specific about supported versions – let's continue this part of the discussion separately.

My assumption is that using `TIMESTAMP` would be more trouble than it's worth if we had to start backporting the implementation to, for example, 6.0.x and up. Is this correct? What do we lose in the long term by using `STRING` indefinitely instead?

Reply:
Thanks for clarifying. Use of the Java client for pull queries requires ksqlDB server version of at least 0.10 (CP 6.0). Are we OK with the migrations tool being incompatible with earlier server versions?

Reply:
Also, is there a reason we'd want to backport this to 6.0.x rather than having users use the tool released with a newer version (even if there server version is 6.0)? There was Java client functionality added post-6.0 (specifically, ksqlDB 0.12) for listing streams/tables that we might like to use in the migrations tool. (It's not a hard requirement so we can avoid it if backporting to 6.0.x is a requirement but I'd like to understand why that would be beneficial.)

Reply:
`CREATE OR REPLACE` was released in 6.1, so wouldn't this tool not even work for 6.0.x?

Reply:
Why do we need `CREATE OR REPLACE`?

Reply:
I guess it would work, but I'd imagine updating schemas would be a major use case.

Reply:
Are you saying that the migrations tool itself needs `CREATE OR REPLACE`, or that users will likely want `CREATE OR REPLACE` in the statements in their migrations files? As long as the migrations tool itself doesn't need `CREATE OR REPLACE`, then the tool is compatible with earlier server versions that do not support `CREATE OR REPLACE`, and can still be used to manage clusters of ksqlDB servers on those older versions. There have been a number of users in the community Slack requesting this feature while operating older ksqlDB server versions so I think that'd still be valuable. You're correct that their sql files wouldn't be able to contain `CREATE OR REPLACE`, but if they're starting from a version that doesn't support it then that should be the case anyway.

Reply:
> The best we can do if we want to use the Java client is keep the tool on master (and 6.1.x if desired) and announce compatibility with servers 6.0.x and later.

I think this is reasonable. Thanks for thinking through the compatibility implications.

Reply:
Thanks, Colin. Sounds like we're all in agreement then.

> What do we lose in the long term by using STRING indefinitely instead?

We don't lose much, just that users might think it's odd that we're storing timestamps in the `STRING` type even though ksqlDB supports `TIMESTAMP`, and we're not creating visibility about the new type for users who don't know about it. This is minor compared to compatibility with older server versions so let's stick with the `STRING` type.

Reply:
Adding to this. Whether we use timestamp or string, I think we should print the string format when displaying time info in the command output. like `ksql-migrations info`.